### PR TITLE
feat: Add Save button to annotations panel

### LINKED ISSUE #1427: Add "Save" button to the title and description overlay
Implement "Save" button as documented in the design refinements from Sprint 5.6. 

Image

### PR DESCRIPTION
**Pull request by bprusinowski**
Closes #1427

This PR adds a "Save" button to the annotations panel. As we save the changes dynamically anyway, it acts as a "close panel" button.

### CODE REVIEW

**Comment by bprusinowski** [comment]
@KerstinFaye I see that in the current design file we also have a `Cancel` button, contrary to what's inside the #1427 issue 😱 

As we have a "live mode editing" across the whole application, where changes are visible on a chart as soon as we make them (except for some data loading related actions, like updating multi-filters where we need to confirm the selection to avoid constant data changes and loading), I wonder if we should change this paradigm for annotations – because if I understand correctly, in the [current design](https://www.figma.com/file/Hn9wvuEYLUmdJtg4KwZYUx/visualize.admin.ch---design?type=design&node-id=3718-229003&mode=design&t=YISUZKohm2MLxOrX-4), user should be able to update title and description without immediate feedback on a chart, but only applied after clicking "Save", and discarded when clicking "Cancel")?

This is a personal preference, but I find it more user-friendly to have this live feedback as we do currently, just wanted to check with you if we should change this (and if so, should we also apply it in other parts of the application)? If you think it's a good idea to change this, maybe then we could also have a small tag above, like "Pending changes", so we know if we actually made a change? I think it might be a bit confusing now, as we can save the changes immediately after entering the panel, without actually updating the title or description yet 👀 

Let me know what you think about this :)

File: app/configurator/components/annotation-options.tsx [hunk 1]
```diff
@@ -92,6 +103,14 @@ const AnnotationOptions = (props: AnnotationOptionsProps) => {
               />
             </Box>
           ))}
+          <Button
+            onClick={handleClosePanel}
+            sx={{ alignSelf: "flex-end", minHeight: 32, mt: 3, px: 5 }}
```
Comment (ptbrowne) [inline app/configurator/components/annotation-options.tsx:108]: Nit: Why do we have to override minHeight and px, this should be handled via the button right ?

File: app/configurator/components/annotation-options.tsx [hunk 2]
```diff
@@ -92,6 +103,14 @@ const AnnotationOptions = (props: AnnotationOptionsProps) => {
...
+          >
+            <Typography component="span">
+              <Trans id="save"> Save</Trans>
```
Comment (ptbrowne) [inline app/configurator/components/annotation-options.tsx:111]:
```suggestion
              <Trans id="save">Save</Trans>
```

**Comment by ptbrowne** [comment]
Code looks good but I am wondering about the behavior here. Here I wonder, there was no modification to how changing a field changes the chart, so Bartosz if I am not mistaken, here in this PR, even if the user does not click "Save", it will have been saved already right ?
I completely agree with Bartosz, the addition of a Save button feels like a change of paradigm. What is exactly the intention there ?

File: app/configurator/components/annotation-options.tsx (hunk 1 above)
Comment (bprusinowski) [reply app/configurator/components/annotation-options.tsx]: Thanks for catching, for height I could use size="small" 👍 For px, it looks different for different buttons in the design currently, see e.g. [this button](https://www.figma.com/file/Hn9wvuEYLUmdJtg4KwZYUx/visualize.admin.ch---design?type=design&node-id=3718-229434&mode=design&t=MBXaeZBQkyiFi3RM-4) and [this button](https://www.figma.com/file/Hn9wvuEYLUmdJtg4KwZYUx/visualize.admin.ch---design?type=design&node-id=3718-229077&mode=design&t=MBXaeZBQkyiFi3RM-4) – in fact maybe this should be minWidth applied in the theme, but I didn't want to do a full review of design to find a rule in this PR, maybe we could check with @KerstinFaye?

**Comment by bprusinowski** [comment]
@ptbrowne yes, it's a "fake" Save button now, as I didn't want to completely change the behavior before discussing this 👍

**Comment by KerstinFaye** [comment]
@bprusinowski Thanks a lot for the feedback. I agree, it's definitely better to have the live feedback. I reviewed the implemented version and I think the solution to have the save button but still the live feedback is best.

**Comment by KerstinFaye** [comment]
@ptbrowne I think this ideas is even based on your feedback that you gave ones in a document that you did for UI improvements. I thought it was a good idea to give the user the opportunity to confirm their action to approve the title input. But if you are all agreeing that it's a change of paradigm then we can also leave it like we had.

File: app/configurator/components/annotation-options.tsx (hunk 1 above)
Comment (KerstinFaye) [reply app/configurator/components/annotation-options.tsx]: [Here are again all Button sizes in the library](https://www.figma.com/file/kYsVD11OnsPAcDGYCHz5jH/Federal-Projects-Global-Library-%E2%80%93%C2%A0CURRENT?type=design&node-id=97-8822&mode=design&t=FYQj7douDXJw5OXL-4)
In this case I would apply the small button size.

**Comment by bprusinowski** [comment]
Thanks for the review @KerstinFaye! From my side it would be fine to leave the current implementation from this PR, let me know what you think @ptbrowne 👍

**Comment by ptbrowne** [comment]
> I think this ideas is even based on your feedback that you gave ones in a document that you did for UI 

Maybe, I have tried to think more about it, and I think my feedback was  to answer to the need of the user "after inputting a title, what do I do now ". I think we want to 

* respect the current paradigm of auto saving (1)
* have an action at the bottom so that the user knows what to do, and does not look around until seeing the close button (2)

I think this PR solution with the "save" button solves (2) but could be confusing to the user, and somehow does not respect (1). So maybe a good solution would be to have the text of the button only "OK".

**Comment by KerstinFaye** [comment]
> I think this PR solution with the "save" button solves (2) but could be confusing to the user, and somehow does not respect (1). So maybe a good solution would be to have the text of the button only "OK".

That works. Let's use "ok".

**Comment by ptbrowne** [comment]
Thanks 👍

**Review by ptbrowne — APPROVED** [review]

**Pull request merged**